core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -694,6 +694,73 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertEquals(outOfRangePos.toLong, outOfRangePartitions.get(tp))
   }
 
+  @Test
+  def testFetchInvalidOffsetResetConfigEarliest(): Unit = {
+    val consumer = createConsumer()
+    this.consumerConfig.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    val totalRecords = 10L
+
+    val producer = createProducer()
+    val startingTimestamp = 0
+    sendRecords(producer, totalRecords.toInt, tp, startingTimestamp = startingTimestamp)
+    consumer.assign(List(tp).asJava)
+    consumeAndVerifyRecords(consumer = consumer, numRecords = totalRecords.toInt, startingOffset =0)
+    // seek to out of range position
+    val outOfRangePos = totalRecords + 1
+    consumer.seek(tp, outOfRangePos)
+    // assert that poll resets to the beginning position
+    consumeAndVerifyRecords(consumer = consumer, numRecords = 1, startingOffset = 0)
+  }
+
+  @Test
+  def testFetchInvalidOffsetResetConfigLatest(): Unit = {
+    val consumer = createConsumer()
+    this.consumerConfig.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest")
```

Comment:
The consumer config should be fed into `createConsumer` method. You can refer to here for how to set config for test consumer: https://github.com/apache/kafka/blob/trunk/core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala#L100-L102

Reply:
thanks! I was following the example in `testFetchInvalidOffset` but you're right I got the order wrong here. I don't know how i missed that I updated the config after building the consumer